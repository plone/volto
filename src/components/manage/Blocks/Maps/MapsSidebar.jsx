import React from 'react';
import { MapsSchema } from './schema';
import { BlockDataForm } from '@plone/volto/components';
import { useIntl, defineMessages } from 'react-intl';
import globeSVG from '@plone/volto/icons/globe.svg';
import { Icon } from '@plone/volto/components';
import { Segment } from 'semantic-ui-react';

const messages = defineMessages({
  Maps: {
    id: 'Maps',
    defaultMessage: 'Maps',
  },
  NoMaps: {
    id: 'No map selected',
    defaultMessage: 'No map selected',
  },
});

const MapsSidebar = (props) => {
  const { data, block, onChangeBlock } = props;
  const intl = useIntl();
  const schema = MapsSchema({ ...props, intl });

  return (
    <>
      {!data.url ? (
        <Segment className="sidebar-metadata-container" secondary>
          {props.intl.formatMessage(messages.NoMaps)}
          <Icon name={globeSVG} size="100px" color="#b8c6c8" />
        </Segment>
      ) : (
        <BlockDataForm
          schema={schema}
          title={intl.formatMessage(messages.Maps)}
          onChangeField={(id, value) => {
            onChangeBlock(block, {
              ...data,
              [id]: value,
            });
          }}
          onChangeBlock={onChangeBlock}
          formData={data}
          block={block}
        />
      )}
    </>
  );
};

export default MapsSidebar;
