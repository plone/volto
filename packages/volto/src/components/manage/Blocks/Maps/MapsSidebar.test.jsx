import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';

import MapsSidebar from './MapsSidebar';

vi.mock('@plone/volto/components/manage/Form', async () => {
  return await import(
    '@plone/volto/components/manage/Form/__mocks__/index.vitest.tsx'
  );
});

const mockStore = configureStore();

test('renders an Image Block Sidebar component', () => {
  const store = mockStore({
    intl: {
      locale: 'en',
      messages: {},
    },
  });
  const component = renderer.create(
    <Provider store={store}>
      <MapsSidebar
        data={{ url: 'map' }}
        block="1234"
        pathname="/news"
        onChangeBlock={() => {}}
        openObjectBrowser={() => {}}
        resetSubmitUrl={() => {}}
      />
    </Provider>,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
