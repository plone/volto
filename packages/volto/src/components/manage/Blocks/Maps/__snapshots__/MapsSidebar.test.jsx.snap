// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an Image Block Sidebar component 1`] = `
<div
  data-schema="{
  \\"title\\": \\"Maps\\",
  \\"block\\": \\"Maps\\",
  \\"fieldsets\\": [
    {
      \\"id\\": \\"default\\",
      \\"title\\": \\"Default\\",
      \\"fields\\": [
        \\"url\\",
        \\"title\\",
        \\"align\\"
      ]
    }
  ],
  \\"properties\\": {
    \\"url\\": {
      \\"title\\": \\"Maps URL\\",
      \\"widget\\": \\"url\\"
    },
    \\"title\\": {
      \\"title\\": \\"Alt text\\"
    },
    \\"align\\": {
      \\"title\\": \\"Alignment\\",
      \\"widget\\": \\"align\\"
    }
  },
  \\"required\\": []
}"
  id="BlockDataForm"
/>
`;
